python/Ganga/GPIDev/Lib/Job/Job.py

```diff
@@ -737,27 +738,28 @@ def outputFilesFailures(self):
 
         return postprocessFailure
 
+    def getSubJobStatuses(self):
+        """
+        This returns a set of all of the different subjob statuses whilst respecting lazy loading
+        """
+
+        if isinstance(self.subjobs, SubJobXMLList):
+            stats = set(self.subjobs.getAllSJStatus())
+        else:
+            stats = set([sj.status for sj in self.subjobs])
```

Comment:
There's no need for the square brackets here. In the case of a function which takes one argument which is a sequence you can write the generator expression directly and omit the nested brackets. It avoids creating a temporary list as well.